src/jit_channel/msgs.rs

```diff
@@ -0,0 +1,263 @@
+			min_fee_msat: self.min_fee_msat,
+			proportional: self.proportional,
+			valid_until: self.valid_until.clone(),
+			min_lifetime: self.min_lifetime,
+			max_client_to_self_delay: self.max_client_to_self_delay,
+			promise,
+		}
+	}
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+/// Fees and parameters for a JIT Channel with promise.
+///
+/// The client will pay max(min_fee_msat, proportional*(payment_size_msat/1_000_000)).
+pub struct OpeningFeeParams {
+	/// The minimum fee required for the channel open.
+	pub min_fee_msat: u64,
+	/// A fee proportional to the size of the initial payment.
+	pub proportional: u32,
+	/// An ISO8601 formatted date for which these params are valid.
+	pub valid_until: String,
+	/// number of blocks that the LSP promises it will keep the channel alive without closing, after confirmation.
+	pub min_lifetime: u32,
+	/// Maximum number of blocks that the client is allowed to set its to_self_delay parameter.
+	pub max_client_to_self_delay: u32,
+	/// Field used by the LSP to validate that these parameters were actually given out by them.
+	pub promise: String,
+}
+
+impl OpeningFeeParams {
+	/// Determine that these parameters are valid given the secret used to generate the promise.
+	// TODO: add validation check that valid_until >= now()
+	pub fn is_valid(&self, promise_secret: &[u8; 32]) -> bool {
+		let mut hmac = HmacEngine::<Sha256>::new(promise_secret);
+		hmac.input(&self.min_fee_msat.to_be_bytes());
+		hmac.input(&self.proportional.to_be_bytes());
+		hmac.input(self.valid_until.as_bytes());
+		hmac.input(&self.min_lifetime.to_be_bytes());
+		hmac.input(&self.max_client_to_self_delay.to_be_bytes());
+		let promise_bytes = Hmac::from_engine(hmac).into_inner();
+		let promise = utils::hex_str(&promise_bytes[..]);
+		promise == self.promise
+	}
+}
+
+/// Information about the parameters a LSP is willing to offer clients
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct GetInfoResponse {
+	/// A set of opening fee parameters.
+	pub opening_fee_params_menu: Vec<OpeningFeeParams>,
+	/// The minimum payment size required to open a channel.
+	pub min_payment_size_msat: u64,
+	/// The maximum payment size the lsp will tolerate.
+	pub max_payment_size_msat: u64,
+}
+
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct BuyRequest {
+	pub version: u16,
+	pub opening_fee_params: OpeningFeeParams,
+	#[serde(skip_serializing_if = "Option::is_none")]
+	pub payment_size_msat: Option<u64>,
+}
+
+/// A response from a buy request made by a client
+///
+/// Includes information needed to construct an invoice.
+#[derive(Clone, Debug, PartialEq, Eq, Deserialize, Serialize)]
+pub struct BuyResponse {
+	/// The short channel id used by LSP to identify need to open channel.
+	pub jit_channel_scid: String,
+	/// The locktime expiry delta the lsp requires.
+	pub lsp_cltv_expiry_delta: u32,
+	/// A flag that indicates who is trusting who.
+	#[serde(default)]
```

Comment:
The spec says this parameter is optional but default to false which is the behavior we get with serde(default).

Reply:
I think we would always write it because it's not possible to construct a BuyResponse without setting it.  It's also currently required for the LSP to provide a value when calling `invoice_parameters_generated`.

I guess we could probably just make it an Option<bool> and whenever it's surfaced to end-user unwrap_or_default() it.  I assume that would allow us to not always write it (set it to None if user passes in false).

I guess I don't really feel strongly, though I'd lean towards not changing it at this point.  It's a bit odd it's optional in the spec, i assume it's just to save a few bytes in the payload?